tide/master/rest/RestController.cpp

```diff
@@ -0,0 +1,194 @@
+{
+    const QByteArray input = QByteArray(data.c_str(), data.size( ));
+    const auto doc = QJsonDocument::fromJson( input );
+
+    if( doc.isNull() || !doc.isObject( ))
+    {
+        put_flog( LOG_INFO, "Error parsing JSON string: '%s'", data.c_str( ));
+        return QJsonObject{};
+    }
+    return doc.object();
+}
+}
+
+RestController::RestController( RestServer& httpServer, DisplayGroup& display )
+    : _displayGroup( display )
+    , _controller( new DisplayGroupController( display ))
+{
+    auto& server = httpServer.get();
+
+    server.handlePUT(_moveToFront);
+    connect ( &_moveToFront, &RestCommand::received,
+              this, &RestController::_handleMoveWindowToFront );
+
+    server.handlePUT(_toggleSelectWindow);
+    connect ( &_toggleSelectWindow, &RestCommand::received,
+              this, &RestController::_handleToggleSelectWindow );
+
+    server.handlePUT(_unfocusWindow);
+    connect ( &_unfocusWindow, &RestCommand::received,
+              this, &RestController::_handleUnfocusWindow );
+
+    server.handlePUT(_moveWindowtoFullscreen);
+    connect ( &_moveWindowtoFullscreen, &RestCommand::received,
+              this, &RestController::_handleMoveWindowToFullscreen );
+
+    server.handlePUT( "tide/resizeWindow", [this]( const std::string& received )
+    { return _handleResizeWindow( received ); } );
+
+    server.handlePUT( "tide/moveWindow", [this]( const std::string& received )
+    { return _handleMoveWindow( received ); } );
+
+    server.handlePUT( "tide/focusWindows", [this]()
+    { return _handleFocusWindows(); });
+
+    server.handlePUT( "tide/unfocusWindows", [this]()
+    { return _handleUnfocusWindows(); });
+
+    server.handlePUT( "tide/exitFullScreen", [this]()
+    { return _handleExitFullScreen(); } );
+
+    server.handlePUT( "tide/deselectWindows", [this]()
+    { return _handleDeselectWindows(); } );
+
+    server.handlePUT( "tide/clear", [this]()
+    { _displayGroup.clear(); return true; } );
+}
+
+bool RestController::_handleMoveWindowToFront( const QString uri )
+{
+    _controller->moveWindowToFront( uri );
+    return true;
+}
+
+bool RestController::_handleFocusWindows()
+{
+    _controller->focusSelected();
+    return true;
+}
+bool RestController::_handleUnfocusWindow( const QString& id )
+{
+   _controller->unfocus( QUuid( id ) );
+    return true;
+}
+bool RestController::_handleUnfocusWindows()
+{
+    _controller->unfocusAll();
+    return true;
+}
+bool RestController::_handleMoveWindowToFullscreen( const QString uri )
+{
+    _controller->showFullscreen( uri );
+    return true;
+}
+bool RestController::_handleExitFullScreen()
+{
+    _controller->exitFullscreen();
+    return true;
+}
+
+bool RestController::_handleResizeWindow( const std::string& string )
+{
+    const auto obj = _toJSONObject( string );
+    if( obj.empty( ))
+        return false;
+
+    const auto windowSize = QSizeF( obj["w"].toString().toDouble(),
```

Comment:
QJsonValue can be directly converted to a double or an int, why make it a string first? Replace all occurences in this file